velox/vector/tests/VectorTest.cpp

```diff
@@ -1671,3 +1671,64 @@ TEST_F(VectorTest, multipleDictionariesOverLazy) {
     ASSERT_EQ(i, dict->as<SimpleVector<int32_t>>()->valueAt(i));
   }
 }
+
+class LazyTestingLoader : public VectorLoader {
+ public:
+  explicit LazyTestingLoader(RowVectorPtr data, int32_t* rowCounter)
+      : data_(data) {
+    rowCounter_ = rowCounter;
+  }
+
+  void loadInternal(RowSet rows, ValueHook* hook, VectorPtr* result) override {
+    (*rowCounter_) += rows.size();
+    *result = data_;
+  }
+
+ private:
+  RowVectorPtr data_;
+  int32_t* rowCounter_;
+};
+
+/// Test lazy loading of dictionary vector
+TEST_F(VectorTest, selectiveLoadingOfLazyDictionary) {
+  auto vectorMaker = std::make_unique<test::VectorMaker>(pool_.get());
+
+  vector_size_t size = 10;
+  auto indices =
+      makeIndices(size, [&](auto row) { return (row % 2 == 0) ? row : size; });
```

Comment:
I used size (explicitly pointing out of bound) to indicate entries that are not captured by the filter because creating a smaller index than the actual values was not allowed. I can get the same behavior by having multiple indices point to the same dictionary value (e.g. 0) I guess. Or, what's the right way to create an index 0,2,4,.. on top of values 0..9?

Reply:
I'm not sure I understand. Would you clarify a bit?

Reply:
Initially the test was set up to create a DictionaryVector with dictionaryValues_ set to [0,1,.., 9] and the indices set to [0,2,..8], but I figured the indices need to be at least the size of the DictionaryVector, right?

  VELOX_CHECK_GE(
      dictionaryIndices->size(),
      length * sizeof(vector_size_t),
      "Malformed dictionary, index array is shorter than DictionaryVector");

[Check in DictionaryVector constructor]

Reply:
>  I figured the indices need to be at least the size of the DictionaryVector, right?

The size of the dictionary vector can be different (smaller or larger) than the size of the base vector. However, size of indices better needs to match (or exceed) size of the dictionary vector. In the case above, we have base vector of size 10 with dictionary vector of size 4. Indices buffer would be size 4 as well. Does that make sense?

Reply:
Got it, let me try to get that into the test.